src/index.ts

```diff
@@ -261,6 +280,21 @@ class CruxPayPeer {
         this._payIDClaim = payIDClaim;
     }
 
+    private _getConfigService = async () => {
+        let configService: BlockstackConfigurationService;
+        const ns: blockstackService.BlockstackService = new blockstackService.BlockstackService({domain: this.walletClientName + CRUX_DOMAIN_SUFFIX});
+        await (this._payIDClaim as PayIDClaim).decrypt();
+        await ns.restoreIdentity((this._payIDClaim as PayIDClaim).virtualAddress as string, {secrets: (this._payIDClaim as PayIDClaim).identitySecrets});
+        const status = await ns.getRegistrationStatus({secrets: (this._payIDClaim as PayIDClaim).identitySecrets});
+        await (this._payIDClaim as PayIDClaim).encrypt();
+        if (status.status === blockstackService.SubdomainRegistrationStatus.DONE) {
+            configService = new BlockstackConfigurationService(this.walletClientName, (this._payIDClaim as PayIDClaim).virtualAddress);
```

Comment:
If I am not wrong, cruxId is being used to set gaia endpoint in the config service, as part of registration status we are anyway fetching gaia data then why refetch again inside BlockstackConfigurationService.

Either we can push registration status check code inside BlockstackConfigurationService or just override the fields which are relevant after fetching gaia endpoint from payIdClaim. As of now this logic seems duplicated.

Reply:
I suppose these two calls are originated using two different types of NameService instances. But not very sure. This is the point where Ankit and Sanchay came to conclusion that the class architecture might be wrong.